pkg/k8shandler/logstore.go

```diff
@@ -6,20 +6,56 @@ import (
   "k8s.io/apimachinery/pkg/api/errors"
   "github.com/sirupsen/logrus"
 
+  "github.com/openshift/cluster-logging-operator/pkg/utils"
+
+  "bytes"
+  "encoding/base64"
+
   metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
   sdk "github.com/operator-framework/operator-sdk/pkg/sdk"
   logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1alpha1"
 )
 
-func CreateOrUpdateLogStore(logging *logging.ClusterLogging)(error) {
-  return nil
+func CreateOrUpdateLogStore(logging *logging.ClusterLogging) error {
+  createOrUpdateESSecret(logging)
+  return createOrUpdateESCR(logging)
+}
+
+func createPasswdFile(user_name string, user_passwd []byte) []byte {
+  var buffer bytes.Buffer
+
+  buffer.WriteString(user_name)
+  buffer.WriteString(":\n  passwd: ")
+  buffer.WriteString(base64.StdEncoding.EncodeToString(user_passwd))
+
+  return buffer.Bytes()
 }
 
-func CreateOrUpdateSecret(logging *logging.ClusterLogging)(error) {
+func createOrUpdateESSecret(logging *logging.ClusterLogging) error {
+
+  esSecret := utils.Secret(
+    "logging-elasticsearch",
+    logging.Namespace,
+    map[string][]byte{
+      "elasticsearch.key": utils.GetFileContents("/tmp/_working_dir/elasticsearch.key"),
+      "elasticsearch.crt": utils.GetFileContents("/tmp/_working_dir/elasticsearch.crt"),
+      "logging-es.key": utils.GetFileContents("/tmp/_working_dir/logging-es.key"),
+      "logging-es.crt": utils.GetFileContents("/tmp/_working_dir/logging-es.crt"),
+      "admin-key": utils.GetFileContents("/tmp/_working_dir/system.admin.key"),
+      "admin-cert": utils.GetFileContents("/tmp/_working_dir/system.admin.crt"),
+      "admin-ca": utils.GetFileContents("/tmp/_working_dir/ca.crt"),
+      "passwd.yml": createPasswdFile("prometheus", utils.GetRandomWord(16)),
+    }  )
+
+  err := sdk.Create(esSecret)
+  if err != nil && !errors.IsAlreadyExists(err) {
+    logrus.Fatalf("Failure constructing Elasticsearch secret: %v", err)
+  }
+
   return nil
 }
 
-func CreateOrUpdateCR(logging *logging.ClusterLogging)(error) {
+func createOrUpdateESCR(logging *logging.ClusterLogging) error {
```

Comment:
for readability consider renaming to expanding `CR` to CustomResource

Reply:
I think to follow the naming pattern I've reused I'll expand `ES` to `Elasticsearch`